www/js/util.ts

```diff
@@ -0,0 +1,93 @@
+import i18next from 'i18next';
+import { DateTime } from 'luxon';
+import humanizeDuration from 'humanize-duration';
+
+/* formatting units for display:
+  - if value >= 100, round to the nearest integer
+   e.g. "105 mi", "119 kmph"
+  - if 10 <= value < 100, round to 1 decimal place
+    e.g. "77.2 km", "11.3 mph"
+  - if value < 10, round to 2 decimal places
+    e.g. "7.27 mi", "0.75 km" */
+export function formatForDisplay(value: number, opts: Intl.NumberFormatOptions = {}): string {
+  if (value >= 100) opts.maximumFractionDigits ??= 0;
+  else if (value >= 10) opts.maximumFractionDigits ??= 1;
+  else opts.maximumFractionDigits ??= 2;
+  return Intl.NumberFormat(i18next.resolvedLanguage, opts).format(value);
+}
+
+/**
+ * @param isoStr1 An ISO 8601 string (with/without timezone)
+ * @param isoStr2 An ISO 8601 string (with/without timezone)
+ * @param opts Intl.DateTimeFormatOptions for formatting (optional, defaults to { weekday: 'short', month: 'short', day: 'numeric' })
+ * @returns A formatted range if both params are defined, one formatted date if only one is defined
+ * @example getFormattedDate("2023-07-14T00:00:00-07:00") => "Jul 14, 2023"
+ * @example getFormattedDate("2023-07-14", "2023-07-15") => "Jul 14, 2023 - Jul 15, 2023"
+ */
+export function formatIso(isoStr1?: string, isoStr2?: string, opts?: Intl.DateTimeFormatOptions) {
```

Comment:
We are using Luxon's `toLocaleString` (which is like a wrapper around the standard `Intl` API's `toLocaleString`), which is doing most of the work here

The functions here exist for 2 reasons:
i) To format date ranges (e.g. `Sep 17, 2024 – Sep 23, 2024`) instead of only a single date (e.g. `Sep 23, 2024`)
ii) Shortcuts so I don't have to pass something like `{ month: 'short', day: 'numeric', year: 'numeric' }` every time